Content.Server/Weapon/Ranged/Systems/GunSystem.cs

```diff
@@ -231,6 +231,11 @@ public void PlayImpactSound(EntityUid otherEntity, DamageSpecifier? modifiedDama
         // 3. Nothing
         var playedSound = false;
 
+        // woops the other entity is deleted
+        // someone needs to handle this better. for now i'm just gonna make it not crash the server -rane
+        if (!HasComp<TransformComponent>(otherEntity))
+            return;
```

Comment:
just explicitly check `Deleted(otherEntity)` instead of roundabout through xform comp